elixir/lib/homework/users/user.ex

```diff
@@ -11,10 +14,42 @@ defmodule Homework.Users.User do
     timestamps()
   end
 
+  @spec changeset(
+          {map, map} | %{:__struct__ => atom | %{__changeset__: map}, optional(atom) => any},
+          :invalid | %{optional(:__struct__) => none, optional(atom | binary) => any}
+        ) :: Ecto.Changeset.t()
   @doc false
   def changeset(user, attrs) do
     user
     |> cast(attrs, [:first_name, :last_name, :dob])
     |> validate_required([:first_name, :last_name, :dob])
+    |> validate_date(:dob)
```

Comment:
Never have unvalidated dates in db.  I stick to a date key (YYYYMMDD) or an ISO/RFC based format in every case, even if the date is system generated.